lib/Sema/CSDiag.cpp

```diff
@@ -2325,6 +2325,51 @@ bool FailureDiagnosis::diagnoseContextualConversionError(
       diagnose(expr->getLoc(), diag::missing_nullary_call, srcFT->getResult())
         .highlight(expr->getSourceRange())
         .fixItInsertAfter(expr->getEndLoc(), "()");
+
+      // It is possible that we're looking at a stored property being
+      // initialized with a closure. Something like:
+      //
+      // var foo: Int = { return 0 }
+      //
+      // Let's offer another fix-it to remove the '=' to turn the stored
+      // property into a computed property. If the variable is immutable, then
+      // replace the 'let' with a 'var'
+
+      // First, check if the variable is declared top level or not. If it is
+      // not, then it means we're inside a decl like a class or an extension.
+      PatternBindingDecl *PBD = nullptr;
+
+      if (auto TLCD = dyn_cast<TopLevelCodeDecl>(CS.DC)) {
```

Comment:
It would be great if you figured out how to detect and diagnose this error via fixes and added fix-it there, I'm trying really hard to remove CSDiag :)

Reply:
@xedin I have an idea - move this code inside `ContextualFailure::diagnoseMissingFunctionCall`. What do you think? For example, we already use `ContextualFailure::trySequenceSubsequenceFixIts` inside this method to apply some fix-its.

Reply:
Sure, it would be great to replace this code with `diagnoseMissingFunctionCall` if possible! Just need to make sure that we actually diagnose it via fix too.

Reply:
@xedin Okay, I have moved it to a fix! Could you take a look?